apps/decoder_csv/lib/decoder/csv.ex

```diff
@@ -26,24 +26,48 @@ defmodule Decoder.Csv do
     def lines_or_bytes(_t), do: :line
 
     def decode(t, messages) do
-      if skip_line?(t) do
-        messages
-        |> Enum.drop(1)
-        |> Enum.map(fn message -> parse(message, t.headers) end)
-      else
-        Enum.map(messages, fn message -> parse(message, t.headers) end)
-      end
+      do_decode(t, messages, t.skip_first_line && !already_skipped?())
```

Comment:
```suggestion
      do_decode(t, messages, t.skip_first_line and !already_skipped?())
```
This would have the same effect but would act as a short-circuit if `t.skip_first_line` evaluates to false and not bother with the second half of the clause.

Reply:
both && and "and" are short circuited.